`-Wmissing-noreturn` error with Clang 21 on macOS

Relevant portion of the build log:

```
[ 13%] Building CXX object examples/all_features/CMakeFiles/disabled.dir/alternative_macros.cpp.o
In file included from /nix/var/nix/b/7y9sy565vxcp6gv88njnvzsxwc/source/examples/all_features/assert_returns_disabled_evaluate.cpp:4:
/nix/var/nix/b/7y9sy565vxcp6gv88njnvzsxwc/source/examples/all_features/assert_returns_disabled.cpp:24:34: error: function 'operator()' could be declared with attribute 'noreturn' [-Werror,-Wmissing-noreturn]
   24 |     if (TEST_FLIP ^ CHECK_THROWS([=] { throw i; }())) { TEST_FAIL(); }
      |                                  ^
/nix/var/nix/b/7y9sy565vxcp6gv88njnvzsxwc/source/examples/all_features/assert_returns_disabled.cpp:25:37: error: function 'operator()' could be declared with attribute 'noreturn' [-Werror,-Wmissing-noreturn]
   25 |     if (TEST_FLIP ^ CHECK_THROWS_AS([=] { throw i; }(), int)) { TEST_FAIL(); }
      |                                     ^
```

--- a/examples/all_features/assert_returns_disabled.cpp
+++ b/examples/all_features/assert_returns_disabled.cpp
@@ -12,6 +12,7 @@ DOCTEST_MAKE_STD_HEADERS_CLEAN_FROM_WARNINGS_ON_WALL_END
 #define TEST_FAIL() std::cout << "FAILED ON: " << __LINE__ \
     << "(" << (TEST_FLIP ? "EVALUATED" : "DISABLED") << ")" << std::endl
 
+DOCTEST_CLANG_SUPPRESS_WARNING_WITH_PUSH("-Wmissing-noreturn")
 static int test_disabled_var_ = [] { // NOLINT
     // none may return true
     if (TEST_FLIP ^ CHECK(0 == 0)) { TEST_FAIL(); }
@@ -28,3 +29,4 @@ static int test_disabled_var_ = [] { // NOLINT
 
     return 0;
 }();
+DOCTEST_CLANG_SUPPRESS_WARNING_POP

